src/manifest/container.rs

```diff
@@ -75,6 +79,238 @@ pub struct Metadata {
     pub version_id: u32,
 }
 
+wire_enum! {
+    /// A hash type for a manifest [`Toc`]
+    ///
+    /// Note that we currently only support the SHA-256 variant, even though
+    /// Cerberus permits SHA-384 and SHA-512 as well.
+    ///
+    /// [`Toc`]: struct.Toc.html
+    #[allow(missing_docs)]
+    pub enum HashType: u8 {
+      Sha256 = 0b000,
+      // Sha384 = 0b001,
+      // Sha512 = 0b010,
+    }
+}
+
+/// A TOC entry's raw bits.
+#[derive(Copy, Clone, PartialEq, Eq, Default, Debug, AsBytes, FromBytes)]
+#[repr(C)]
+struct RawTocEntry {
+    element_type: u8,
+    format_version: u8,
+    offset: u16,
+    len: u16,
+    parent_idx: u8,
+    hash_idx: u8,
+}
+
+/// An entry to a manifest's table of contents.
+///
+/// A TOC entry describes an element in manifest, such as its format and its
+/// location.
+///
+/// A TOC entry is encoded exactly the same way as this struct is laid
+/// out, byte-for-byte.
+///
+/// See [`Toc`](struct.Toc.html).
+pub struct TocEntry<'entry, 'toc, M> {
+    toc: &'entry Toc<'toc, M>,
+    // Invariant: index is always a valid index into `toc.entries`.
+    index: usize,
+    _ph: PhantomData<fn() -> M>,
```

Comment:
With your `toc` taking type parameter `M` is this needed?

Reply:
Nope.